### PR TITLE
feat: add support for JSON arrays

### PR DESCRIPTION
**Pull request by aliehsaeedii**
### Description 
This FEAT converts a string with JSON array to an array of JSON strings - JSON_ITEMS(String) -> Array[String]

Then, a user could do something like

TRANSFORM(JSON_ITEMS(message), (obj) => EXTRACTJSONFIELD(obj, '$.time`))

### Testing done 
Unit test
QTT

### Reviewer checklist
see #9062

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/json/JsonItems.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  public List<String> items(@UdfParameter final String jsonItems) {
+    if (jsonItems == null) {
+      return null;
+    }
+
+
+    final List<JsonNode> objectList = UdfJsonMapper.readAsJsonArray(jsonItems);
+    final List<String> res = new ArrayList<>();
+    for (JsonNode jsonNode: objectList) {
+      if (jsonNode.isMissingNode() || !jsonNode.isObject()) {
```
Comment (Gerrrr) [inline ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/json/JsonItems.java:48]: Why don't we support all JSON types here?

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/json_items.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        {"topic": "test_topic", "key": "1", "value": {"events": "[{\"type\": \"A\", \"timestamp\": \"2022-01-27\"}, {\"type\": \"B\", \"timestamp\": \"2022-05-18\"}]"}},
+        {"topic": "test_topic", "key": "1", "value": {"events": "[]"}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": "1", "value": {"events": ["{\"type\":\"A\",\"timestamp\":\"2022-01-27\"}", "{\"type\":\"B\",\"timestamp\":\"2022-05-18\"}"]}},
+        {"topic": "OUTPUT", "key": "1", "value": {"events": []}}
+      ]
+    },
+    {
+      "name": "extract timestamps from json objects",
```
Comment (Gerrrr) [inline ksqldb-functional-tests/src/test/resources/query-validation-tests/json_items.json:22]: Thank you for adding an integration test that covers the end-to-end usage of this function!

**Review by Gerrrr — COMMENTED** [review]
Looks good overall! Can you please update the `scalar_functions.md` doc before merging? Also, please see ksqlDB guidelines on commit messages https://github.com/confluentinc/ksql/blob/master/CONTRIBUTING.md#commit-messages.

**Comment by vcrfxia** [comment]
Hey @aliehsaeedii , can you update the title for this PR when you get a chance? With the current title, the feature will not appear in the auto-generated changelog. For a user-facing feature like this, it'd be great to have it included. See more here: https://github.com/confluentinc/ksql/blob/master/CONTRIBUTING.md#commit-messages

**Review by Gerrrr — APPROVED** [review]
Looks good! Please don't forget to adjust the commit message according to the guideline before merging.

**Pull request merged**